File: backend/internal/models/semester.go
package models

type Semester struct {
	OrgID         int64  `json:"org_id"`
	ClassroomID   int64  `json:"classroom_id"`
	OrgName       string `json:"org_name"`
	ClassroomName string `json:"classroom_name"`
	Active        bool   `json:"active"`
}


type GetSemester struct {
  SemesterID  int64  `json:"semester_id"`
}


type SemesterID struct {
	ID int64 `json:"semester_id"`
}

File: frontend/src/types/semester.ts
export interface Semester {
  org_id: number;
  classroom_id: number;
  org_name: string;
  classroom_name: string;
  active: boolean;
}

//TODO: refactor this to come in as a map of {orgid: Semester[]}
export interface UserSemestersResponse {
  active_semesters: Semester[];
  inactive_semesters: Semester[];
}

export interface OrgSemestersResponse {
  semesters: Semester[];
}
